integration/secret_cert_test.go

```diff
@@ -127,3 +129,130 @@ func TestCertificateUpload(t *testing.T) {
 
 	testenv.Test(t, f)
 }
+
+func TestCertificateUploadForOAS(t *testing.T) {
+	var (
+		tlsSecretName = "test-oas-cert-upload-secret"
+
+		testNS string
+		tykOas *v1alpha1.TykOasApiDefinition
+		tykEnv environment.Env
+		tykCtx context.Context
+	)
+	f := features.New("test certificate upload").
+		Setup(func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+			eval := is.New(t)
+			var err error
+
+			testNS, _ = ctx.Value(ctxNSKey).(string) // nolint: errcheck
+
+			// Obtain Environment configuration to be able to connect Tyk.
+			tykEnv, err = generateEnvConfig(ctx, envConf)
+			eval.NoErr(err)
+
+			res := semver.Compare(tykEnv.TykVersion, "v4.1")
+			if res < 0 {
+				t.Skip("OAS support is added in Tyk in v4.1")
+			}
+
+			tykCtx = tykClient.SetContext(context.Background(), tykClient.Context{
+				Env: tykEnv,
+				Log: log.NullLogger{},
+			})
+
+			// Create TLS secret
+			_, err = createTestTlsSecret(ctx, testNS, envConf, func(s *v1.Secret) {
+				s.Name = tlsSecretName
+			})
+			eval.NoErr(err)
+
+			tykOASDoc := `{
+				"info": {
+				  "title": "Petstore",
+				  "version": "1.0.0"
+				},
+				"openapi": "3.0.3",
+				"components": {},
+				"paths": {},
+				"x-tyk-api-gateway": {
+				  "info": {
+					"name": "Petstore",
+					"state": {
+					  "active": true
+					}
+				  },
+				  "upstream": {
+					"url": "https://petstore.swagger.io/v2"
+				  },
+				  "server": {
+					"listenPath": {
+					  "value": "/petstore/",
+					  "strip": true
+					},
+					"clientCertificates":{
+					  "enabled": true
```

Comment:
Does `clientCertificates` field needs to be set? What happens if users specify `clientCertificateRef` field in CRD while the OAS JSON file in ConfigMap lacks of 
```json
"clientCertificates":{"enabled": true}
```

I now that enabled needs to be set to true explicitly but i am wondering what happens if there is no `clientCertificates` field in OAS ConfigMap

Reply:
It is still a valid API Definition. We support MTLS via 2 flows. 
1. static MTLS: Here is certificate is provided in the API Definition itself
2. dynamic MTLS: User can provide certificate in the key, upload keys through developer portal 
Reference to docs https://tyk.io/docs/basic-config-and-security/security/mutual-tls/client-mtls/

Reply:
Yes, that will be valid but i would like to know whether operator will be able to parse and populate non-existing field in the JSON object.

As far as I am concerned, `OASSetClientCertificates` handles non existing keys but would like to double check with you

Reply:
It handles it.